ehcache-impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -994,59 +1002,54 @@ private long getSizeOfKeyValuePairs(K key, OnHeapValueHolder<V> holder) throws L
 
     @IgnoreSizeOf
     private final Supplier<ValueHolder<V>> source;
```

Comment:
I think you should be able to remove the source reference here, and instead have the original initiating thread hold the source on-stack, call it, and then complete the Fault appropriately.

Reply:
After a thorough check, finds that ye, we can remove this -` Supplier<ValueHolder<V>> source` attribute from `Fault `class
Accommodated you review comment